yargs.js

```diff
@@ -301,42 +301,58 @@ function Yargs (processArgs, cwd, parentRequire) {
     return self
   }
 
-  self.demand = self.required = self.require = function (keys, max, msg) {
+  self.demandOption = self.demand = self.required = self.require = function (keys, max, msg) {
```

Comment:
I am wondering if we should still keep `self.demand` as a fallback (?) for `demandOption`, or just completely rename it to `demandOption`